openmdao/core/total_jac.py

```diff
@@ -155,17 +155,15 @@ def __init__(self, problem, of, wrt, use_abs_names, return_format, approx=False,
         driver_wrt = list(driver._designvars)
         driver_of = driver._get_ordered_nl_responses()
 
-        # Convert of and wrt names from promoted to absolute
+        # In normal use, of and wrt always contain variable names. However, there are unit tests
+        # that don't specify them, so we need these here.
         if wrt is None:
-            if driver_wrt:
-                prom_wrt = driver_wrt
-            else:
-                raise RuntimeError("Driver is not providing any design variables "
```

Comment:
I'm not following why you removed the empty dv and empty response checks here and replaced them with checks in two different places (in compute_totals and check_totals).

Reply:
Fixed

Reply:
This is what we talked about. Now, the of and wrt are always passed in to this init.  No reason to check them here anymore, since it should never happen.

Reply:
It would happen if you weren't doing the upper level checks, wouldn't it?

Reply:
I guess the checks were checking for of/wrt == None, so you're right, that wouldn't happen.